dietpi/dietpi-software

```diff
@@ -17651,6 +17653,67 @@ List of installed software and their online documentation URLs:
 			# Load all automation vars
 			FirstRun_Automation_Init
 
+			if (( $AUTOINSTALL_RESTORE )); then
+
+					# Find unmounted drives
+					lsblk -rnpo NAME,UUID,MOUNTPOINT | mawk '$2 && ! $3 {print $1}' > .lsblk_out_tmp
+					i=0
+					while read -r line
+					do
+
+						# Mount drives to temporary mount points
+						mkdir "/mnt/dietpi-backup$i"
+						mount "$line" "/mnt/dietpi-backup$i"
+
+						((i++))
+
+					done < .lsblk_out_tmp
+					rm .lsblk_out_tmp
+
+					mapfile -t alist < <(find /mnt -type f -name '.dietpi-backup_stats')
+
+					if [[ $AUTOINSTALL_RESTORE == 1 ]]; then
+						# Do we have any results?
+						if [[ ${alist[0]} ]]; then
+							# Create List for Whiptail
+							G_WHIP_MENU_ARRAY=()
+							for i in "${alist[@]}"
+							do
+
+								last_backup_date=$(sed -n '/ompleted/s/^.*: //p' "$i" | tail -1) # Date of last backup for this backup
+								backup_directory=${i%/.dietpi-backup_stats} # Backup directory (minus the backup file), that we can use for target backup directory.
+								G_WHIP_MENU_ARRAY+=("$backup_directory" ": $last_backup_date")
+
+							done
+
+							G_WHIP_MENU 'Please select a previous backup to use:'
+							/boot/dietpi/dietpi-backup -1 "$G_WHIP_RETURNED_VALUE"
+
+						else
+							G_WHIP_MSG 'No previous backups were found in /mnt/*. Install will continue like normal.'
+						fi
+					elif [[ $AUTOINSTALL_RESTORE == 2 ]]; then
+						# Do we have any results?
+						if [[ ${alist[0]} ]]; then
+							# Restore first found backup
+							/boot/dietpi/dietpi-backup -1 /mnt/dietpi-backup0
+						else
+							G_DIETPI-NOTIFY 1 'DietPi-Backup auto-restore was selected but no backup has been found in /mnt/*. The dietpi.txt option will be downgraded to 1 (interactive restore).'
+							# Downgrade dietpi.txt option
+							sed -i 's/[[:blank:]]*AUTO_SETUP_BACKUP_RESTORE=2/AUTO_SETUP_BACKUP_RESTORE=1/' /boot/dietpi.txt
+						fi
+					fi
+
+				# Remove mounted drives
+				umount /mnt/dietpi-backup[0-9]*
+				rmdir /mnt/dietpi-backup[0-9]*
+
+				# Boot into normal DietPi
+				echo 2 > /boot/dietpi/.install_stage
```

Comment:
That file was restored with the backup as well. Probably we should skip that step in case intentionally a different install stage is restored? Or do I oversee something that makes necessary?